### PR TITLE
DEV-2239 Make some Tax attributes optional

### PR DESCRIPTION
**Pull request by juanmex**

### CODE REVIEW

File: spec/buho_cfdi/nodes/tax_transferred_spec.rb [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
-
-  context 'required params' do
-    it { should validate_presence_of(:base) }
-    it { should validate_presence_of(:tax) }
-    it { should validate_presence_of(:factor_type) }
+RSpec.describe ::Nodes::TaxTransferred, type: :model do
+  context 'when factor_type is not Exento' do
+    let (:node) { ::Nodes::TaxTransferred.new({tax: '001', factor_type: 'Tasa', base: '100', rate_or_fee: '0.1000', import: '100'}) }
+    it 'has all values' do
+      expect(node.valid?).to eq(true)
```
Comment (yusent) [inline spec/buho_cfdi/nodes/tax_transferred_spec.rb]:
```suggestion
      expect(node).to be_valid
```

File: spec/buho_cfdi/nodes/tax_transferred_spec.rb [hunk 2]
```diff
@@ -1,19 +1,27 @@
...
+    end
   end
 
-  describe '#to_hash' do
-    it 'is expected to return a kind of Hash' do
-      expect(subject.to_hash).to be_kind_of(Hash)
+  context 'when factor_type is Exento' do
+    let (:node) { ::Nodes::TaxTransferred.new({tax: '001', factor_type: 'Exento', base: '100'}) }
+    it 'has only required values' do
+      expect(node.valid?).to be(true)
```
Comment (yusent) [inline spec/buho_cfdi/nodes/tax_transferred_spec.rb]:
```suggestion
      expect(node).to be_valid
```

File: spec/buho_cfdi/nodes/tax_transferred_spec.rb [hunk 3]
```diff
@@ -1,19 +1,27 @@
...
+      expect(node.tax).to eq('001')
+      expect(node.factor_type).to eq('Exento')
+      expect(node.base).to eq('100')
+      expect(node.rate_or_fee).to eq(nil)
```
Comment (yusent) [inline spec/buho_cfdi/nodes/tax_transferred_spec.rb]:
```suggestion
      expect(node.rate_or_fee).to be_nil
```

File: spec/buho_cfdi/nodes/tax_transferred_spec.rb [hunk 4]
```diff
@@ -1,19 +1,27 @@
...
+      expect(node.import).to eq(nil)
```
Comment (yusent) [inline spec/buho_cfdi/nodes/tax_transferred_spec.rb]:
```suggestion
      expect(node.import).to be_nil
```

File: spec/buho_cfdi/nodes/transferred_spec.rb [hunk 5]
```diff
@@ -1,19 +1,27 @@
...
-  context 'required params' do
-    it { should validate_presence_of(:tax) }
-    it { should validate_presence_of(:factor_type) }
-    it { should validate_presence_of(:rate_or_fee) }
-    it { should validate_presence_of(:import) }
+RSpec.describe ::Nodes::Transferred, type: :model do
+  context 'when factor_type is not Exento' do
+    let (:node) { ::Nodes::Transferred.new({tax: '001', factor_type: 'Tasa', base: '100', rate_or_fee: '0.1000', import: '100'}) }
+    it 'has all values' do
+      expect(node.valid?).to eq(true)
```
Comment (yusent) [inline spec/buho_cfdi/nodes/transferred_spec.rb]:
```suggestion
      expect(node).to be_valid
```

File: spec/buho_cfdi/nodes/transferred_spec.rb [hunk 6]
```diff
@@ -1,19 +1,27 @@
...
+    end
   end
 
-  describe '#to_hash' do
-    it 'is expected to return a kind of Hash' do
-      expect(subject.to_hash).to be_kind_of(Hash)
+  context 'when factor_type is Exento' do
+    let (:node) { ::Nodes::Transferred.new({tax: '001', factor_type: 'Exento', base: '100'}) }
+    it 'has only required values' do
+      expect(node.valid?).to be(true)
```
Comment (yusent) [inline spec/buho_cfdi/nodes/transferred_spec.rb]:
```suggestion
      expect(node).to be_valid
```

File: spec/buho_cfdi/nodes/transferred_spec.rb [hunk 7]
```diff
@@ -1,19 +1,27 @@
...
+      expect(node.tax).to eq('001')
+      expect(node.factor_type).to eq('Exento')
+      expect(node.base).to eq('100')
+      expect(node.rate_or_fee).to eq(nil)
+      expect(node.import).to eq(nil)
```
Comment (yusent) [inline spec/buho_cfdi/nodes/transferred_spec.rb]:
```suggestion
      expect(node.import).to be_nil
```

File: spec/buho_cfdi/nodes/transferred_spec.rb [hunk 8]
```diff
@@ -1,19 +1,27 @@
...
-    it 'is expected to return a kind of Hash' do
-      expect(subject.to_hash).to be_kind_of(Hash)
+  context 'when factor_type is Exento' do
+    let (:node) { ::Nodes::Transferred.new({tax: '001', factor_type: 'Exento', base: '100'}) }
+    it 'has only required values' do
+      expect(node.valid?).to be(true)
+      expect(node.tax).to eq('001')
+      expect(node.factor_type).to eq('Exento')
+      expect(node.base).to eq('100')
+      expect(node.rate_or_fee).to eq(nil)
```
Comment (yusent) [inline spec/buho_cfdi/nodes/transferred_spec.rb]:
```suggestion
      expect(node.rate_or_fee).to be_nil
```

File: spec/buho_cfdi/nodes/tax_transferred_spec.rb (hunk 1 above)
Comment (juanmex) [reply spec/buho_cfdi/nodes/tax_transferred_spec.rb]: All my changes are in 14ea7dcb31609090f517cdcff8298a0f98c7dd5c

**Comment by juanmex** [comment]
All suggestions were applied in 14ea7dcb31609090f517cdcff8298a0f98c7dd5c

**Review by ococom1476 — APPROVED** [review]

**Review by yusent — APPROVED** [review]

**Pull request merged**